### DISCUSSION TITLE
window is not defined in Solid starter project

### DISCUSSION THREAD

**Discussion by jcubic** [solidjs/solid#2157] [Q&A] [answered]
I created a solid TypeScript project with:

```bash
npm init solid@latest
Need to install the following packages:
create-solid@0.5.8
Ok to proceed? (y) y
┌  
 Create-Solid v0.5.8
│
◇  Project Name
│  Test
│
◇  Is this a Solid-Start project?
│  Yes
│
◇  Which template would you like to use?
│  basic
│
◇  Use Typescript?
│  Yes
│
◇  Project successfully created! 🎉
│
◇  To get started, run: ─╮
│                        │
│  cd Test               │
│  npm install           │
│  npm run dev           │
│                        │
├────────────────────────╯
```

And when I wanted to access the window object got an error that it was not defined. Does default solid project use SSR? How to indicate that the component should be run on the client?

I tried to use `globalThis.window` but this gave errors in other parts of the code when it attempted to assign value to window, which was undefined.

I'm trying to execute this at the top level of the app.tsx file:

```typescript
import $ from 'jquery';
import terminal from 'jquery.terminal';
import 'jquery.terminal/css/jquery.terminal.min.css';

terminal(window, $);
```

**Comment by jcubic** [comment, marked as answer]
It seems that I created a [Solid Start](https://github.com/solidjs/solid-start) project.

**Reply by kimutaiRop** [reply]
so anything that needs window cannot be used in solid start or what? am confused thought even with solidstart you should be able to use things like chartjs which require window.

am saying this coz I see `◇  Is this a Solid-Start project?` from the create project flow

**Reply by fdemb** [reply]
You *can* use things that need window, but only in client-side context. Solid Start executes both on the server (where `window` isn't defined), and in the browser. A good way of dealing with this is https://docs.solidjs.com/solid-start/reference/client/client-only
